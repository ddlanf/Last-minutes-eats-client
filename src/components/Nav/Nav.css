.nav-bar{
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    width: 100%;
    margin: 0;
    height: 6vw;
    max-height: 100px;
    z-index: 10;
    
}

.nav-bar-left-nav, .nav-bar-right-nav{
    position: absolute;
    text-align: left;
    font-size: 15px;
}


.nav-bar-left-nav{
    padding-left: 3%;
    padding-top: 4px;
    width: 130px;
}

.nav-bar-right-nav{
    right: 0;
    padding-top: 15px;
    padding-right: 3%;
}

a.nav-bar-right-nav-link, a.nav-bar-left-nav-link{
    color: white;
    text-decoration: none;
}

.nav-break{
    display: block;
    margin-left: 23%;
}

.logo-container{
    width: 100%;
    display: flex;
    justify-content: center;
    text-align: center;
}

.line{
    width: 100%;
    min-height: 50px;
    height: 13vw;
    max-height: 50px;
    background-color: #002D52;
    position: absolute;
    z-index: -10;
    
}

.get-started{
    margin-right: 11%;
    opacity: 0;
    animation: appear 1s 5s forwards;
}

.get-started-arrow{
    margin-left: 25px;
    font-size: 50px;
    color: black;
    margin-bottom: 0;
    position: relative;
    animation: upAndDown 1s linear 3s infinite;
}

.get-started-text{
    width: 60%;
    margin-left: 0%;
    margin-top: -5px;
    text-align: center;
    font-size: 12px;
    color: blue;
    font-weight: bold;
}

@keyframes upAndDown {
    0% { top: -5px}
    50% { top: 0px}
    100% { top: -5px}
}

@keyframes appear {
    0% { opacity: 0;}
    100% { opacity: 1;}
}

img.last-minute-eats-logo{
    margin-top: 1.6px;
    width: 15vw;
    max-width: 70px;
}

@media screen and (min-width: 1000px) {

    
    .nav-bar-left-nav, .nav-bar-right-nav{
        font-size: 28px;
        padding-top: 20px;
    }
    
    .nav-break{
        display: inline;
    }
    
    .nav-bar-left-nav{
        padding-left: 3%;
        padding-top: 30px;
        width: 300px;
    }
    
    .nav-bar-right-nav{
        right: 0;
        padding-top: 30px;
        padding-right: 3%;
    }
    
    .nav-break{
        display: inline;
        margin-left: 6px;
    }

    img.last-minute-eats-logo{
        margin-top: 1.6px;
        width: 12vw;
        max-width: 150px;
    }
    
    .line{
        min-height: 50px;
        height: 8vw;
        max-height: 100px;
    }
    
    .get-started-arrow{
        position: relative;
        margin-top: 10px;
        left: 80px;
        font-size: 5vw;
        position: relative;
        animation: upAndDown 1s linear 3s infinite;
    }
    
    .get-started-text{
        width: 500px;
        margin-left: 0%;
        text-align: center;
        font-size: 30px;
        position: relative;
        bottom: 10px;
    }
}