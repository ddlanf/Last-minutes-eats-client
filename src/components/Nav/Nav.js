import React, { Component } from 'react'
import { Link, Route } from 'react-router-dom'
import Logo from '../Logo/Logo'
import "./Nav.css"

class Nav extends Component {

    rendorRightNavBar(){
        const rightNavBar = 
        ['view-all-recipes', 'view-recipe/:recipeId', 'edit-recipe/:recipeId']
        .map(path =>{
           
           return(  
                    <Route
                         exact key={path} path={`/${path}`}>
                        <Link className="nav-bar-right-nav-link" to="/make-recipe">Make Recipe</Link>
                    </Route>);
        })

        rightNavBar.push(
            (<Route
                exact key={'/'} path={`/`}>
                {this.props.hasEmailToken ?
                    '': <Link className="nav-bar-right-nav-link" to="/signup">Sign Up</Link>}
            </Route>)
        )

        return rightNavBar
    }


    render() {

        return (
            <nav className="nav-bar">
                <div className="nav-bar-left-nav">
                    <Link 
                        className="nav-bar-left-nav-link"
                        to='/view-all-recipes'
                        >
                        Last Minute 
                            <span className="nav-break">Eats</span>
                    </Link>
                    <Route  exact path="/">
                        <div className="get-started">
                            <div className="get-started-arrow">â†‘</div>
                            <p className="get-started-text">Click here to get started!</p>
                        </div>
                    </Route>
                </div>
                <div className="logo-container">
                    <div className="line"></div>
                    <img className="last-minute-eats-logo" src={require('../../images/logo.png')} alt="logo"/>
                </div>
                <div className="nav-bar-right-nav">
                    {this.rendorRightNavBar()}
                </div>
            </nav>
        )
    }
}


export default Nav