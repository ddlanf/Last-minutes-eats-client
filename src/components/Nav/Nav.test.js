import React from 'react';
import ReactDOM from 'react-dom';
import Nav from './Nav';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import rootReducer from '../../reducers/index'


describe('Make component', () => {
    
  const store = createStore(rootReducer, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());

    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                      <Provider store={store}>
                        <Nav/>
                      </Provider>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(
                <BrowserRouter>
                  <Provider store={store}>
                    <Nav/>
                  </Provider>
                </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});