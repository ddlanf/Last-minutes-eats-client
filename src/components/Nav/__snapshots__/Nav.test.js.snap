// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make component renders the UI as expected 1`] = `
Array [
  <section
    className="sign-up"
  >
    <form
      className="sign-up-form "
      onAnimationEnd={[Function]}
      onSubmit={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    >
      <label
        className="sign-up-label"
      >
        Please enter your email to get early access to the full version of the App.
      </label>
      <input
        className="sign-up-input"
        name="email"
        onChange={[Function]}
        type="text"
      />
      
      
      <button
        className="sign-up-submit"
        disabled={false}
        type="submit"
      >
        Submit
      </button>
      <button
        className="sign-up-back"
        onClick={[Function]}
        type="button"
      >
        Back
      </button>
    </form>
  </section>,
  <nav
    className="nav-bar"
  >
    <div
      className="nav-bar-left-nav"
    >
      <a
        className="nav-bar-left-nav-link"
        href="/view-all-recipes"
        onClick={[Function]}
      >
        Last Minute
        <span
          className="nav-break"
        >
          Eats
        </span>
      </a>
      <div
        className="get-started"
      >
        <div
          className="get-started-arrow"
        >
          â†‘
        </div>
        <p
          className="get-started-text"
        >
          Click here to get started!
        </p>
      </div>
    </div>
    <div
      className="logo-container"
    >
      <div
        className="line"
      />
      <a
        className="last-minute-eats-logo"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="not-loading-recipes"
          src="logo.png"
        />
      </a>
    </div>
    <div
      className="nav-bar-right-nav"
    >
      <button
        className="nav-bar-right-nav-link"
        onClick={[Function]}
      >
        Sign Up
      </button>
    </div>
  </nav>,
]
`;
